includes/abilities-api/class-wp-abilities-category-registry.php

```diff
@@ -0,0 +1,246 @@
+	 *                                  alphanumeric characters and dashes.
+	 * @param array<string,mixed> $args An associative array of arguments for the category. See wp_register_ability_category() for
+	 *                                  details.
+	 * @return ?\WP_Ability_Category The registered category instance on success, null on failure.
+	 *
+	 * @phpstan-param array{
+	 *   label?: string,
+	 *   description?: string,
+	 *   ...<string, mixed>
+	 * } $args
+	 */
+	public function register( string $slug, array $args ): ?WP_Ability_Category {
+		if ( ! doing_action( 'abilities_api_categories_init' ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				sprintf(
+					/* translators: 1: abilities_api_categories_init, 2: category slug. */
+					esc_html__( 'Categories must be registered during the %1$s action. The category %2$s was not registered.' ),
+					'<code>abilities_api_categories_init</code>',
+					'<code>' . esc_html( $slug ) . '</code>'
+				),
+				'n.e.x.t'
+			);
+			return null;
+		}
+
+		if ( $this->is_registered( $slug ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				/* translators: %s: Category slug. */
+				esc_html( sprintf( __( 'Category "%s" is already registered.' ), $slug ) ),
+				'n.e.x.t'
+			);
+			return null;
+		}
+
+		if ( ! preg_match( '/^[a-z0-9]+(?:-[a-z0-9]+)*$/', $slug ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				esc_html__( 'Category slug must contain only lowercase alphanumeric characters and dashes.' ),
+				'n.e.x.t'
+			);
+			return null;
+		}
+
+		/**
+		 * Filters the category arguments before they are validated and used to instantiate the category.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param array<string,mixed> $args The arguments used to instantiate the category.
+		 * @param string              $slug The slug of the category.
+		 */
+		$args = apply_filters( 'register_ability_category_args', $args, $slug );
+
+		try {
+			// WP_Ability_Category::prepare_properties() will throw an exception if the properties are invalid.
+			$category = new WP_Ability_Category( $slug, $args );
+		} catch ( \InvalidArgumentException $e ) {
+			_doing_it_wrong(
+				__METHOD__,
+				esc_html( $e->getMessage() ),
+				'n.e.x.t'
+			);
+			return null;
+		}
+
+		$this->registered_categories[ $slug ] = $category;
+		return $category;
+	}
+
+	/**
+	 * Unregisters a category.
+	 *
+	 * Do not use this method directly. Instead, use the `wp_unregister_ability_category()` function.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @see wp_unregister_ability_category()
+	 *
+	 * @param string $slug The slug of the registered category.
+	 * @return ?\WP_Ability_Category The unregistered category instance on success, null on failure.
+	 */
+	public function unregister( string $slug ): ?WP_Ability_Category {
+		if ( ! $this->is_registered( $slug ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				/* translators: %s: Category slug. */
+				sprintf( esc_html__( 'Category "%s" not found.' ), esc_attr( $slug ) ),
```

Comment:
Small detail, but might be important for gaining the context when translating in the interface disconnected from the code:

```suggestion
				/* translators: %s: Ability category slug. */
				sprintf( esc_html__( 'Ability category "%s" not found.' ), esc_attr( $slug ) ),
```